app/routes.py

```diff
@@ -1,2 +1,205 @@
+        task.description = form_data["description"]
+        task.is_complete = form_data["completed_at"] 
+
+        db.session.commit()
+        return jsonify({"task": task.json_object()}),200
+
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+        return jsonify({"details": f'Task {task.id} "{task.title}" successfully deleted'}), 200
+
+@task_bp.route("", methods=["GET", "POST"])
+def handle_tasks():
+    if request.method == "GET":
+        
+        request_value = request.args.get("sort") 
+        
+        if request_value == None:
+            tasks = Task.query.all()
+
+        if request_value == "asc":
+            tasks = Task.query.order_by(Task.title.asc()) 
+        
+        
+        if request_value == "desc":
+            tasks = Task.query.order_by(Task.title.desc())
+        task_response = []
+
+        for task in tasks:
+        
+            task_response.append(task.json_object())
+        
+        return jsonify(task_response), 200
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+
+        if "completed_at" not in request_body or "description" not in request_body or "title" not in request_body:
+            return jsonify({
+                "details": "Invalid data"
+            }), 400
+
+        new_task = Task(title=request_body["title"],
+                        description=request_body["description"],
+                        completed_at=request_body["completed_at"])
+
+        db.session.add(new_task)
+        db.session.commit()
+        
+        return jsonify({"task": new_task.json_object()}), 201
+
+        db.session.add(new_task)
+        db.session.commit()
+
+def slack_bot(task):
+    url = "https://slack.com/api/chat.postMessage?channel=task-notifications"
+    slack_token = os.environ.get("SLACK")
+
+    payload = {"text": f"Someone just completed the task: '{task.title}'!"}
+    headers = {"Authorization": f"Bearer {slack_token}"}
+
+    return requests.request("PATCH", url, headers=headers, data=payload)
+
+
+@task_bp.route("/<task_id>/mark_complete", methods=["PATCH"], strict_slashes=False)
+def mark_complete(task_id):
+    
+    if request.method == "PATCH":
+        
+        task = Task.query.get(task_id)
+        
+        if task is None:
+            return jsonify(None), 404
+        
+        task.completed_at = datetime.now()
+
+    
+        db.session.commit()
+        
+        # call slackbot
+        slack_bot(task)
+
+        
+        return jsonify({"task": task.json_object()}),200 
+
+
+@task_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"], strict_slashes=False)
+def mark_incomplete(task_id):
+    
+    if request.method == "PATCH":
+        
+        task = Task.query.get(task_id)
+        
+        if task is None:
+            return jsonify(None), 404
+        
+        task.completed_at = None
+        
+        return jsonify({"task":task.json_object()}),200
+    
+
```

Comment:
careful with adding too many blank lines! it can make code harder to read

```suggestion
    if request.method == "PATCH":
        task = Task.query.get(task_id)
        
        if task is None:
            return jsonify(None), 404
        
        task.completed_at = None
        
        return jsonify({"task":task.json_object()}),200
```